exporter/datadogexporter/internal/testutil/logs.go

```diff
@@ -90,3 +93,32 @@ func gUnzipData(rg io.Reader) ([]byte, error) {
 	}
 	return io.ReadAll(r)
 }
+
+// MockLogsEndpoint returns the processed JSON log data for each endpoint call
+func MockLogsEndpoint(w http.ResponseWriter, r *http.Request) JsonLogs {
+	// we can reuse same response object for logs as well
+	res := metricsResponse{Status: "ok"}
+	resJSON, err := json.Marshal(res)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		log.Fatalln(err)
+	}
+	req, err := gUnzipData(r.Body)
+	if err != nil {
+		w.WriteHeader(http.StatusBadRequest)
+		log.Fatalln(err)
+	}
+	var jsonLogs JsonLogs
+	err = json.Unmarshal(req, &jsonLogs)
+	if err != nil {
```

Comment:
I can't apply this pattern for the line that uses `gUnzipData(r.Body)` because I need to use the result of this late ron in the function. Would it make sense to abstract this if statement to the helper function?